tsl/src/compression/create.c

```diff
@@ -307,7 +312,7 @@ compresscolinfo_init(CompressColInfo *cc, Oid srctbl_relid, List *segmentby_cols
 		{
 			cc->col_meta[colno].algo_id = 0; // invalid algo number
 		}
-		coldef = makeColumnDef(NameStr(attr->attname), attroid, -1 /*typmod*/, 0 /*collation*/);
+		coldef = makeColumnDef(NameStr(attr->attname), attroid, typmod, collid);
```

Comment:
wouldnt this change require a migration otherwise definitions of compressed hypertables differ before and after the change?

Reply:
They do change. I think the migration is not feasible and can't be automated (e.g. you'll have to recreate the indexes if you have any). The rest of the code should still work with the old compressed chunk definitions. So the bug is only fixed for the newly created chunks.

Reply:
And the old chunks will continue working, but they'll still have the bug.